great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -2021,11 +2027,15 @@ def _sqlalchemy_column_map_condition_values(
             message=f'Error: The column "{column_name}" in BatchData does not exist.'
         )
 
-    query = (
-        sa.select([sa.column(column_name).label("unexpected_values")])
-        .select_from(selectable)
-        .where(unexpected_condition)
+    query = sa.select([sa.column(column_name).label("unexpected_values")]).where(
+        unexpected_condition
     )
+    if not MapMetricProvider.is_sqlalchemy_metric_selectable(map_metric_provider=cls):
+        query = (
+            sa.select([sa.column(column_name).label("unexpected_values")])
+            .where(unexpected_condition)
+            .select_from(selectable)
+        )
```

Comment:
I find this clearer as an `if-else` with the first block moved below into `else`.